### ISSUE TITLE
Quick open is much slower

### ISSUE THREAD

**Issue by josh-tepper** [Microsoft/vscode#63873]
Issue Type: <b>Bug</b>

When I use the quick open/command palette to open a file (ie -- by typing the file name directly into the command pallette), the results render much slower than in previous versions.  It used to be the case that I would see filenames as I typed, with results refreshing immediately.  Following the latest monthly update, this is not the case.  Often after I finish typing I can count several seconds before seeing the file name results!

I have the following setting turned off:
```
"search.quickOpen.includeSymbols": false,
```

Not sure whether or not this issue is a duplicate or not; see also the issue below.  It seems that it might refer to something else:
https://github.com/Microsoft/vscode/issues/60419

VS Code version: Code 1.29.1 (bc24f98b5f70467bc689abf41cc5550ca637088e, 2018-11-15T19:06:21.742Z)
OS version: Darwin x64 17.7.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz (8 x 2800)|
|GPU Status|2d_canvas: enabled<br>checker_imaging: disabled_off<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: enabled<br>rasterization: enabled<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 2, 2|
|Memory (System)|16.00GB (0.18GB free)|
|Process Argv|-psn_0_495737|
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (19)</summary>

Extension|Author (truncated)|Version
---|---|---
language-x86-64-assembly|13x|2.2.11
ActiveFileInStatusBar|Ros|1.0.3
better-toml|bun|0.3.2
solargraph|cas|0.17.6
gitignore|cod|0.6.0
gitlens|eam|8.5.6
remotehub|eam|0.2.0
nasc-touchbar|fel|1.11.0
vscode-pull-request-github|Git|0.2.3
jsonnet|hep|0.1.0
git-tree-compare|let|1.4.0
python|ms-|2018.10.1
cpptools|ms-|0.20.1
csharp|ms-|1.17.1
Go|ms-|0.7.0
vscode-docker|Pet|0.4.0
ruby|reb|0.20.0
vscode-open-in-github|sys|1.10.0
vscode-proto3|zxh|0.2.2

(2 theme extensions excluded)

</details>

**Comment by josh-tepper** [comment]
looks to be related:
https://github.com/Microsoft/vscode/issues/63857
https://github.com/Microsoft/vscode/issues/63840

**Comment by roblourens** [comment]
Can you try downloading the Insiders' build and tell me whether it's any better there now? https://code.visualstudio.com/insiders

**Comment by josh-tepper** [comment]
Nope -- it even took so long to show file results that I was able to grab a screenshot (not unusual on the new build):
![screen shot 2018-12-06 at 8 58 25 pm](https://user-images.githubusercontent.com/20231989/49622992-ca0d9380-f999-11e8-99a2-0cccee13ba73.png)

However, one interesting observation -- after waiting for the results to finally load, I can edit the text in the pallet and the file results update nearly instantly.  I can even delete all the text and type a new string (results, still updating quickly as I type).  I just can't leave the command pallet and go back.  Once I do that multi-second waiting happens again!

**Comment by roblourens** [comment]
Is this a very large workspace? Do you see a process named `rg` running when the progress bar is running?

Is it possible that you have lots of symlinks in your workspace, or `search.useIgnoreFiles` disabled, or something like that?

**Comment by josh-tepper** [comment]
It is a large workspace, but I do have `search.userIgnoreFiles` enabled.

I would tell you about `rg` except that I can't seem to be able to see the process table while this is happening!  ps hangs and both top and the builtin vscode process explorer stops updating!  As soon as the results finish populating, ps returns accurately and top/vscode process explorer return to normal.  I have literally never seen this before and I have no idea whats going on.  I don't know enough about how on OS X these process read the process table in order to even speculate about this.

**Comment by Sebioff** [comment]
Same problem here, before 1.29 the results showed up pretty much instantly without a noticeable delay. Since 1.29 it takes 20-30 seconds for any results to show up when starting Code for the first time after a system reboot (Windows 10). After that it is usually quite fast again but sometimes still with a longer delay that I can't recall noticing before 1.29.
The workspace contains ~1.500 C# files and is stored on an M.2 SSD.
No settings have been changed by me between 1.29 and the previous version.

I can provide further info if needed but am not familiar with the internal workings and debug tools of VS Code, so detailed instructions would be required.

**Comment by josh-tepper** [comment]
I see the exact same behavior when running rg directly -- the process table stops updating, and it takes about 10s to run, and the process does not respond to a sigstop -- implying that it's in uninterruptible sleep.  The very interesting thing is:

This behavior presents itself whether or not I invoke rg correctly with the `--files` argument, or without any arguments causing it to print the usage error message.  In both cases the process runs for > 10s!
- A "real" run -- arguments `--file`, outputs the list of files, and returns with status code 0:
   ```
   real	0m10.688s
   user	0m0.000s
   sys	0m0.005s
   ```
 - Providing no arguments resulting in a usage error message, and returning status code 1:
   ```
   real	0m10.686s
   user	0m0.008s
   sys	0m0.012s
    ```
I'm going to try to trace the system calls to see what's happening!

**Comment by roblourens** [comment]
That is very strange! I wonder whether you see anything similar with a `grep -R`? I'm very curious to see what is blocking it. FYI @BurntSushi

**Comment by BurntSushi** [comment]
@roblourens Hmm, yeah I don't think I have any ideas either, although it's a bit hard to follow this issue. Perhaps someone could describe a reproduction in more detail. Otherwise, it's difficult to know what command is actually being run, on which inputs and what the output is.

In my limited experience, I recall that the Windows anti-malware scanner (I think?) slowing things down considerably. I disabled it long ago in my test setup. Maybe something has changed with respect to that? Just taking a shot in the dark.

**Comment by roblourens** [comment]
This is on Mac, so antimalware shouldn't be an issue here. All I know here is that it seems that the rg process takes a very long time to complete, even when it does no real work "Providing no arguments resulting in a usage error message" and has some weird side effects on the system: "process table stops updating"

**Comment by Sebioff** [comment]
For me quick open is as fast as it used to be again since the 1.30.0 update.

**Comment by BurntSushi** [comment]
Oh, then yeah, I have no idea. I don't think I could even venture a guess. At this point, I'd probably start looking at syscalls (on Linux, I'd use `strace`, I don't know what the equivalent is on macOS). Is there a syscall that's being blocked? If so, that might be insightful. But otherwise, ripgrep isn't doing anything weird here AFAIK.

**Comment by josh-tepper** [comment]
Ok -- I got to the bottom of this:  TL;DR -- In the end it was anti-virus software, but figuring this out involved a bit of digging:
- I used dtruss (a dtrace based MacOS utility) to trace the syscalls.  And shockingly, even though the process took over 10s to run, it seemed that the program was only actually executing for a few ms.
- I wrote "hello world" in c -- and same behavior -- it took 10.6s to run, but the program was only executing for 1.5ms
- I suspected that this might have something to do with the way processes are forked, so I wrote another c program to fork and run my hello world, and traced that -- and that's where I saw something very strange.  The syscall execve (which loads a program) ran for about 10.6s!
- At first I was stumped: what has access to the kernel and can interfere with something as basic as the loading of a process?  Then my coworker pointed out that this is exactly what our antivirus was doing to scan binaries (via some integration with the kernel).  I turned it off -- everything is better.

**Comment by BurntSushi** [comment]
@josh-tepper Interesting find! Do you happen to know why this problem started happening recently but didn't occur previously?

**Comment by roblourens** [comment]
Thanks for troubleshooting and following up! Mac antivirus, that's interesting. What's it called?

**Comment by josh-tepper** [comment]
No idea.  But I can say that vscode wasn't the only affected program.  I had a huge headache with the chrome 71 update, docker, postman, etc.  It seems that my system was increasingly affected over the last three weeks. It wasn't clear that it was all related until I turned off the antivirus tool and it solved everything.  We're still investigating the root cause.

**Comment by roblourens** [comment]
Can't get viruses if you can't use your computer 👍
